src/main/scala/logger/SymbExLogger.scala

```diff
@@ -232,256 +231,177 @@ object SymbExLogger {
     * @param pcs    Current path conditions.
     */
   @elidable(INFO)
-  def openMemberScope(member: ast.Member, s: State, pcs: PathConditionStack): Unit = {
-    memberList = memberList ++ Seq(new SymbLog(member, s, pcs))
-  }
+  def openMemberScope(member: ast.Member, pcs: PathConditionStack): Log = {
+    try {
+      val log = newEntityLogger(member, pcs)
+
+      synchronized {
+        if(isClosed) {
+          // If we time out and close the log, but somehow manage to race to move on to a new member in time, the log is
+          // closed so the records are not vacuously recorded, and the log is not added to the map of logs.
+          log.close()
+        } else {
+          members += member -> log
+        }
+      }
 
-  /** Use this method to access the current log, e.g., to access the log of the method
-    * that gets currently symbolically executed.
-    *
-    * @return Returns the current method, predicate or function that is being logged.
-    */
-  def currentLog(): SymbLog = {
-    if (enabled)
-      memberList.last
-    else NoopSymbLog
+      log.openMemberScope()
+      log
+    } catch {
+      case anything: Throwable =>
+        println(anything)
```

Comment:
Silicon uses a proper logging framework. Please use that instead.

Reply:
Oh this is just a debugging remnant, I don't think there is a concrete reason to catch or log exceptions here.